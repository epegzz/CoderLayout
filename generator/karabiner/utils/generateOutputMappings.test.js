const generateOutputMappings = require('./generateOutputMappings')

describe('generateOutputMappings', () => {
  test('with triggers', () => {
    const karabinerConfig = {
      triggerKeys: ['x', 'y', 'v', 'w'],
      layers: {
        layer1: {
          // 'x' or 'y'
          triggerKeys: ['x', 'y'],
          mapping: [
            { from: 'a', to: [{ keyName: 'b', modifiers: []}] }
          ]
        },
        layer2: {
          // 'v' and 'w'
          triggerKeys: [['v', 'w']],
          mapping: [
            { from: 'c', to: [{ keyName: 'd', modifiers: []}] }
          ]
        }
      }
    }

    expect(generateOutputMappings(karabinerConfig)).toMatchSnapshot()
  })

  test('without triggers', () => {
    const karabinerConfig = {
      layers: {
        layer1: {
          mapping: [
            { from: 'a', to: [
              { keyName: 'b', modifiers: []},
              { keyName: 'b', modifiers: ['shift']}
            ] }
          ]
        }
      }
    }

    expect(generateOutputMappings(karabinerConfig)).toMatchSnapshot()
  })

  test('with modifiers', () => {
    const karabinerConfig = {
      layers: {
        layer1: {
          mapping: [
            { from: 'a', to: [{ keyName: 'b', modifiers: ['left_command']}] }
          ]
        }
      }
    }

    expect(generateOutputMappings(karabinerConfig)).toMatchSnapshot()
  })

  test('disabling keys', () => {
    const karabinerConfig = {
      layers: {
        layer1: {
          mapping: [
            { from: 'a', to: [] }
          ]
        }
      }
    }

    expect(generateOutputMappings(karabinerConfig)).toMatchSnapshot()
  })

  test('non keylayout mapping', () => {
    const karabinerConfig = {
      layers: {
        layer1: {
          mapping: [
            { from: 'a', to: [{ keyName: 'left_shift', modifiers: []}] },
            { from: 'b', to: [{ keyName: 'left_control', modifiers: []}] },
            { from: 'c', to: [{ keyName: 'left_command', modifiers: []}] },
            { from: 'd', to: [{ keyName: 'left_option', modifiers: []}] },
          ]
        }
      }
    }

    expect(generateOutputMappings(karabinerConfig)).toMatchSnapshot()
  })

})
