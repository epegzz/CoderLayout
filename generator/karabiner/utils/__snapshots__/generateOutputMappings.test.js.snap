// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateOutputMappings disabling keys 1`] = `
Array [
  Object {
    "description": "Layer layer1",
    "manipulators": Array [
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "a",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
    ],
  },
]
`;

exports[`generateOutputMappings non keylayout mapping 1`] = `
Array [
  Object {
    "description": "Layer layer1",
    "manipulators": Array [
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "a",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "left_shift",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "b",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "left_control",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "c",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "left_command",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "d",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "left_option",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
    ],
  },
]
`;

exports[`generateOutputMappings with modifiers 1`] = `
Array [
  Object {
    "description": "Layer layer1",
    "manipulators": Array [
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "a",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "b",
            "modifiers": Array [
              "left_command",
            ],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
    ],
  },
]
`;

exports[`generateOutputMappings with triggers 1`] = `
Array [
  Object {
    "description": "Layer layer1",
    "manipulators": Array [
      Object {
        "conditions": Array [
          Object {
            "name": "trigger_x",
            "type": "variable_if",
            "value": 1,
          },
          Object {
            "name": "trigger_y",
            "type": "variable_unless",
            "value": 1,
          },
          Object {
            "name": "trigger_v",
            "type": "variable_unless",
            "value": 1,
          },
          Object {
            "name": "trigger_w",
            "type": "variable_unless",
            "value": 1,
          },
        ],
        "from": Object {
          "key_code": "a",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "b",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
      Object {
        "conditions": Array [
          Object {
            "name": "trigger_y",
            "type": "variable_if",
            "value": 1,
          },
          Object {
            "name": "trigger_x",
            "type": "variable_unless",
            "value": 1,
          },
          Object {
            "name": "trigger_v",
            "type": "variable_unless",
            "value": 1,
          },
          Object {
            "name": "trigger_w",
            "type": "variable_unless",
            "value": 1,
          },
        ],
        "from": Object {
          "key_code": "a",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "b",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
    ],
  },
  Object {
    "description": "Layer layer2",
    "manipulators": Array [
      Object {
        "conditions": Array [
          Object {
            "name": "trigger_v",
            "type": "variable_if",
            "value": 1,
          },
          Object {
            "name": "trigger_w",
            "type": "variable_if",
            "value": 1,
          },
          Object {
            "name": "trigger_x",
            "type": "variable_unless",
            "value": 1,
          },
          Object {
            "name": "trigger_y",
            "type": "variable_unless",
            "value": 1,
          },
        ],
        "from": Object {
          "key_code": "c",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "d",
            "modifiers": Array [],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
    ],
  },
]
`;

exports[`generateOutputMappings without triggers 1`] = `
Array [
  Object {
    "description": "Layer layer1",
    "manipulators": Array [
      Object {
        "conditions": Array [],
        "from": Object {
          "key_code": "a",
          "modifiers": Object {
            "mandatory": Array [],
            "optional": Array [
              "any",
            ],
          },
        },
        "to": Array [
          Object {
            "key_code": "b",
            "modifiers": Array [],
          },
          Object {
            "key_code": "b",
            "modifiers": Array [
              "shift",
            ],
          },
        ],
        "to_after_key_up": Array [],
        "to_if_alone": Array [],
        "type": "basic",
      },
    ],
  },
]
`;
